src/hddsupport.c

```diff
@@ -213,14 +202,95 @@ void hddLoadModules(void)
             LOG("[XHDD]:\n");
             sysLoadModuleBuffer(&xhdd_irx, size_xhdd_irx, 0, NULL);
         }
+
         if (ret < 0) {
             LOG("HDD: No HardDisk Drive detected.\n");
             setErrorMessageWithCode(_STR_HDD_NOT_CONNECTED_ERROR, ERROR_HDD_IF_NOT_DETECTED);
             return;
         }
+    } else
+        hddModulesLoadCount++;
+
+    LOG("HDDSUPPORT LoadModules done\n");
+}
+
+// Returns 1 for MBR/GPT, 0 for APA, and -1 if an error occured
```

Comment:
This is also very questionable function. Id rather drop that check completely as there many false positive cases here. I think it is safe to rely on bdm driver, if it can load the game list, it is fine, if it cannot, nothing bad happens. But by using this function, some setups will exit early, while being possible to be loaded by BDM.

Reply:
This function is pretty much just a fail safe/idiot proof function.. if a user turns on apa hdd setting while using an exFat hdd without this function OPL will load apa/pfs modules and likely corrupt the drive by trying to create partitions and files; with this function it’ll just bail out and not load the modules, games list won’t populate but it won’t break any drives, leading the user to try the other hdd setting

Reply:
ok, I see the point